completion_aggregator/serializers.py

```diff
@@ -282,6 +282,65 @@ def get_mean(self, obj):
         return mean
 
 
+class CourseCompletionStatsSerializer(serializers.Serializer):
+    """
+    Serialize completions stats at the course level.
+    """
+
+    course_key = serializers.CharField()
+    completion = _CompletionSerializer(source='*')
+    username = serializers.SerializerMethodField()
+    mean = serializers.SerializerMethodField()
+
+    optional_fields = {'mean', 'username'}
+
+    def __init__(self, instance, requested_fields=frozenset(), *args, **kwargs):
+        """
+        Initialize a course completion stats serializer.
+
+        Add any requested optional fields.
+        """
+        super(CourseCompletionStatsSerializer, self).__init__(
+            instance, *args, **kwargs)
+        for field in self.optional_fields - requested_fields:
+            del self.fields[field]
+
+    def get_username(self, obj):
+        """
+        Serialize the username.
+        """
+        return obj.user.username
+
+    def _calculate_mean(self, obj):
+        """
+        Caclulate mean completion percent for all enrolled users.
+        """
+        enrollments = compat.get_users_enrolled_in(obj.course_key)
+        enrollment_count = enrollments.count()
+        if enrollment_count == 0:
+            return 0.
+
+        total = Aggregator.objects.filter(
+            course_key=obj.course_key,
+            aggregation_name='course',
+        ).aggregate(
+            total=Sum(Coalesce('percent', Value(0.)))
+        ).get('total') or 0.
+
+        return total / enrollment_count
```

Comment:
All this work is already performed in the view.  It looks like you're throwing it out and redoing the calculation, but without the filtering.  At this point, `mean` should just be available as a field on `obj`.